src/Raven.Server/Documents/Indexes/LastProcessedReference.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections.Generic;
+using Raven.Server.Documents.Indexes.Workers;
+
+namespace Raven.Server.Documents.Indexes
+{
+    public class LastProcessedReference
+    {
+        private readonly Dictionary<(string Collection, string ReferencedDocumentId), (long ReferencedDocumentEtag, string ItemId)> _lastIdPerCollectionForDocuments = new Dictionary<(string, string), (long, string)>();
```

Comment:
In this case, won't we cause a memory leak? What will clean up old values here?
In general, I think we are better off having a class here, which will hold all those properties.
There is a single instance per collection, no?

Reply:
I clear it after a successful References batch.
https://github.com/ravendb/ravendb/pull/11301/files#diff-37a2d054996e0ffcdc07cf1e752cbc4d6c5b9babbe0ba563f577ab7b119adfb8R328